### PR TITLE
Add getLevelName method in extension

### PR DESCRIPTION
**Pull request by aminalaee**
- Adds `_getLevelName` to the extension code.
- Exposes `getLevelName` from `picologging` instead of `logging`.
- Fixes the `Logger.__repr__` to show effective level instead of level and  be in sync with the logging style:
https://github.com/python/cpython/blob/c485ec014ce174bb3f5ae948151dc40e0f6d5f7f/Lib/logging/__init__.py#L1831

### CODE REVIEW

File: src/picologging/_picologging.cxx [hunk 1]
```diff
@@ -12,8 +13,39 @@
 #include "handler.hxx"
 #include "streamhandler.hxx"
 
+std::map<unsigned short,std::string> LEVELS_TO_NAMES = {
```
Comment (aminalaee) [inline src/picologging/_picologging.cxx:16]: One question/point is that if the [addLevelName](https://docs.python.org/3/library/logging.html#logging.addLevelName) is added, will it be ok to assign to this map as I guess it's not thread-safe. If that's ever going to be done...

File: tests/unit/test_logger.py [hunk 2]
```diff
@@ -196,7 +195,7 @@ def test_logger_init_bad_args():
 
 def test_logger_repr():
     logger = picologging.Logger("goo", picologging.DEBUG)
-    assert repr(logger) == "<Logger 'goo' (10)>"
+    assert repr(logger) == "<Logger 'goo' (DEBUG)>"
```
Comment (tonybaloney) [inline tests/unit/test_logger.py:198]: Please add tests for the following: 

1. `__repr__` for invalid level no doesn't crash
2. `getLevelName()` for non integer returns TypeError
3. `getLevelName()` returns correct value for known level number

File: src/picologging/_picologging.cxx (hunk 1 above)
Comment (tonybaloney) [reply src/picologging/_picologging.cxx:16]: use `unordered_map` instead of `map` since the order is irrelevant.

You can also mark this as `const`

**Review by tonybaloney — CHANGES_REQUESTED** [review]

File: src/picologging/_picologging.cxx (hunk 1 above)
Comment (aminalaee) [reply src/picologging/_picologging.cxx]: Done.

File: tests/unit/test_logger.py (hunk 2 above)
Comment (aminalaee) [reply tests/unit/test_logger.py]: Added tests for item 1 and 2 and item 3 was already done there in `test_picologging.py`.

**Comment by tonybaloney** [comment]
Nice! thanks for the PR

**Pull request merged**